pkg/controller/core/workload_controller.go

```diff
@@ -263,35 +283,56 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 
+		var requeueAfter *time.Duration
+		if wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt != nil {
+			requeueAfter = ptr.To(wl.Status.RequeueState.RequeueAt.Sub(r.clock.Now()))
+		}
+		if requeueAfter != nil && *requeueAfter > 0 {
+			log.V(3).Info("Waiting for backoff to finish", "backoff", *requeueAfter)
+			return reconcile.Result{RequeueAfter: *requeueAfter}, nil
+		}
+
+		if requeueAfter != nil && workload.Status(&wl) == workload.StatusPending {
+			log.V(3).Info("Pending workload requeued after backoff")
+
+			err := workload.PatchAdmissionStatus(ctx, r.client, &wl, r.clock, func() (*kueue.Workload, bool, error) {
+				if wl.Status.RequeueState == nil || wl.Status.RequeueState.RequeueAt == nil {
+					return nil, false, nil
+				}
+				wl.Status.RequeueState.RequeueAt = nil
+				if wl.Status.RequeueState.Count == nil {
+					wl.Status.RequeueState = nil
+				}
+				workload.ResetAdmissionCheckRequeueState(&wl)
+				return &wl, true, nil
+			})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return ctrl.Result{}, err
+			}
+
+			if err := r.queues.AddOrUpdateWorkload(wl.DeepCopy()); err != nil {
+				log.V(2).Info("ignored an error for now", "error", err)
```

Comment:
```suggestion
				log.V(2).Info("failed to put the workload back into queue", "error", err)
```
Feel free to rephrase or drop the log
this might actually be a bugfix because it could error and we would silence that. cc @PBundyra